### DISCUSSION TITLE
Libre2 + Novopen6 + xDrip: any experience?

### DISCUSSION THREAD

**Discussion by dagord** [NightscoutFoundation/xDrip#2873] [Q&A]
I'm really trying to understand how xDrip is supposed to work with Libre2 sensors (provided that I'm based in Italy so I'm using the european sensor version).
My girlfirend is happily using Libre2 + Miaomiao3 + xDrip (via OOP2) but she's going to switch to a phone with NFC support, so she'll should be able to get rid of Miaomiao.
I understand that a patched Freestyle Libre app is required, but it's not clear how this works (according to some websites one need to scan the sensor once with the FLS app, then disable it and scan the sensor again with xDrip; but according to official xDrip docs you can use FSL to get readings from the sensor and then xDrip to get readings from FSL).
Please note also that my girlfriend will use Novopen6 too, as suggested by her diabetologist, and it should be supported by xDrip...
Anyone using this setup could share his experience?

**Comment by Navid200** [comment]
@psonnera

**Comment by psonnera** [comment]
You can get rid of miaomiao and use direct connection to xDrip+. You don't need the patched app. The only difficulty is to steal bluetooth communication from the vendor app. The original documentation is here https://www.minimallooper.com/post/how-to-setup-freestyle-libre-2-and-oop2-to-use-a-native-bluetooth-connection-in-xdrip and also here https://xdrip.readthedocs.io/en/latest/install/libre2/. Novopen should be supported now, together with Libre NFC, but I'm not 100% sure.

**Reply by dagord** [reply]
Hello and thanks for your answer.
I've already read the minimallooper walkthrough but, since is dated march 2021, I didn't know it was still valid.

I'm still very confused so I'm try to explain why (plaese be patient) :)

1. https://www.minimallooper.com/post/how-to-setup-freestyle-libre-2-and-oop2-to-use-a-native-bluetooth-connection-in-xdrip says I need to scan the sensor with the modified FSL, then disable the app and scan it again with xDrip

2. On the other hand, https://xdrip.readthedocs.io/en/latest/troubleshoot/libre2/ says

> xDrip+ will automatically receive data from the patched app using the specific hardware data source [Libre2 (patched App)](https://xdrip.readthedocs.io/en/latest/install/libre2patch/)

According to this, it seems the FSL app is required since it actually gets data from the sensor and broadcasts them to xDrip (this is what I understand).
So it seems they are referring to two different apps, or two different ways to use the same app.

3. The docs you linked (https://xdrip.readthedocs.io/en/latest/install/libre2/) also say that the initial scan can be accomplished with the standalone reader too, so no app would be required at all

These are conflicting informations to me... 

Please note that my girlfriend is not a "geek person" at all, so I will accomplish the initial setup but she must be as independent as possible on the daily usage!

Thanks again and sorry for being verbose.

**Reply by psonnera** [reply]
Try that for now https://www.glicemiadistanza.it/libre-2-e-xdrip-collegamento-diretto/ and I'll elaborate the rest of the answer this evening.

**Reply by dagord** [reply]
Yesterday my girlfriend came home with the new smartphone (Redmi Note 12), so I decided to go with a quick try following the last website you linked: it worked like a charm! (the sensor was started a couple of days ago with the standalone FSL reader)
I encountered only a minor issue: the walkthrough forgots to say that the sensor have to be NFC scanned again after being activated; by the way this was mentioned on the minimallooper website.
Thank you very much!
